components/CommonModalComponent.tsx

```diff
@@ -0,0 +1,351 @@
+    if (show) {
+      request();
+      // eslint-disable-next-line no-restricted-globals
+      history.pushState(null, "", location.href);
+      window.addEventListener("popstate", modalOff);
+      document.body.style.cssText = `
+      position: fixed;
+      top: -${window.scrollY}px;
+      overflow-y: scroll;
+      width: 100%;`;
+    }
+    return () => {
+      const scrollY = document.body.style.top;
+      document.body.style.cssText = "";
+      window.scrollTo(0, parseInt(scrollY || "0", 10) * -1);
+      window.removeEventListener("popstate", modalOff);
+    };
+  }, [show]);
+
+  const customTitle = useMemo(() => {
+    if (type === "folder_rename" || type === "folder_add") {
+      return (
+        <FolderHeader>
+          <h2 className="modal_header">{title}</h2>
+        </FolderHeader>
+      );
+    }
+    if (type === "folder_share" || type === "folder_delete") {
+      return (
+        <FolderHeader>
+          <h2 className="modal_header">{title}</h2>
+          <p className="sub_title">{folderName}</p>
+        </FolderHeader>
+      );
+    }
+    if (type === "link_delete" || type === "into_folder") {
+      return (
+        <FolderHeader>
+          <h2 className="modal_header">{title}</h2>
+          <p className="sub_title">{linkName}</p>
+        </FolderHeader>
+      );
+    }
+  }, [type, title, folderName, linkName]);
+
+  const info = useMemo(() => {
+    if (type === "folder_add") {
+      return (
+        <FolderArticle>
+          <input placeholder="내용 입력" />
+          <AddButton>추가하기</AddButton>
+        </FolderArticle>
+      );
+    }
+    if (type === "folder_rename") {
+      return (
+        <FolderArticle>
+          <input placeholder="내용 입력" defaultValue={folderName} />
+          <AddButton>변경하기</AddButton>
+        </FolderArticle>
+      );
+    }
+    if (type === "folder_share") {
+      return (
+        <ShareArticle>
+          <div>
+            <button
+              className="link_button"
+              style={{ backgroundColor: "#FEE500" }}
+            >
+              <img src="/images/modal_kakao.png" />
+            </button>
+            <p>카카오톡</p>
+          </div>
+          <div>
+            <button
+              className="link_button"
+              style={{ backgroundColor: "#1877F2" }}
+            >
+              <img src="/images/modal_facebook.png" />
+            </button>
+            <p>페이스북</p>
+          </div>
+          <div>
+            <button className="link_button">
+              <img src="/images/modal_link.png" />
+            </button>
+            <p>링크복사</p>
+          </div>
+        </ShareArticle>
+      );
+    }
+    if (type === "folder_delete") {
+      return <DeleteButton>삭제하기</DeleteButton>;
+    }
+    if (type === "link_delete") {
+      return <DeleteButton>삭제하기</DeleteButton>;
+    }
+    if (type === "into_folder") {
+      return (
+        <>
+          <IntoArticle>
+            {folderData?.data?.map((e, i) => {
```

Comment:
이런식으로 표현해주는게 좋을 것 같네요!
```suggestion
            {folderData?.data?.map((folder, index) => {
```